app/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -284,45 +239,53 @@ export default class DateTimePicker extends React.Component {
     )
   }
 
-  renderHours() {
-    const hours = []
+  renderHours(displayDateTime) {
+    const selectOptions = []
+    const selectValues = []
+    const currentTimeLabel = displayDateTime.format('HH:mm')
+
+    let valueIndex = -1
 
     for (let i = 0; i < 24 * this.TIME_PICKER_HOUR_SUBDIVISIONS; i++) {
       const date = this.getInternalDate({
         hours: 0,
-        minutes: i * 30
+        minutes: i * 30,
+        seconds: 0
       })
+      const label = new DateTime(date).format('HH:mm')
 
-      hours.push(
-        <li key={date.getTime()}>
-          <button
-            className={styles.hour}
-            onClick={this.handleDatePick.bind(this, date)}
-            ref={element => {
-              this.hoursRefs[i] = {
-                date,
-                element
-              }
-            }}
-            type="button"
-          >
-            {new DateTime(date).format('HH:mm')}
-          </button>
-        </li>
-      )
+      selectOptions.push({
+        label,
+        value: i
+      })
+
+      if (label === currentTimeLabel) {
+        valueIndex = i
+      }
+
+      // `Select` can only hold string values, so we need to store the actual
+      // Date objects in a separate array and then retrieve them by index.
+      selectValues.push(date)
+    }
+
+    // If `valueIndex` is -1, it means the current time isn't one of the set
+    // intervals (e.g. 19:43). In this case, we add that value as an option
+    // to <Select>, and make it both selected and disabled.
+    if (valueIndex === -1) {
+      selectOptions.unshift({
+        disabled: true,
+        label: currentTimeLabel,
+        value: valueIndex
+      })
     }
 
     return (
-      <div className={styles['hours-container']}>
-        <ul
-          className={styles.hours}
-          ref={element => {
-            this.hoursContainerRef = element
-          }}
-        >
-          {hours}
-        </ul>
-      </div>
+      <Select
+        dir="up"
+        onChange={e => this.handleDatePick(selectValues[e.target.value])}
```

Comment:
I think I would strongly discourage creating *more* functions that are garbage collected on every render; it's just one small thing each time, but they add up

Reply:
In this case it's a bit more involved, because the event handler needs access to the `selectValues` object we construct above.

Reply:
Ah, of course, I'm an idiot